https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/commit/dd4dc5e8bce2c03631c3613dbddee1a691bdd17d

From dd4dc5e8bce2c03631c3613dbddee1a691bdd17d Mon Sep 17 00:00:00 2001
From: Jan Palus <jpalus@fastmail.com>
Date: Fri, 17 Jun 2022 14:36:36 +0200
Subject: [PATCH] bluetooth/gst: Correct var type for GST_TYPE_BITMASK

GST_TYPE_BITMASK is 64-bit bit mask while corresponding channel_mask in
pulseaudio is int therefore usually 32-bit. Switch to uint64_t instead
to match internal representation in gstreamer.

Fixes pulseaudio crash on ARM 32-bit when pulseaudio is compiled with
gstreamer and either LDAC or aptX support is available.

Part-of: <https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/723>
---
 src/modules/bluetooth/a2dp-codec-gst.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/modules/bluetooth/a2dp-codec-gst.c b/src/modules/bluetooth/a2dp-codec-gst.c
index 8ef74be9c..11839c580 100644
--- a/src/modules/bluetooth/a2dp-codec-gst.c
+++ b/src/modules/bluetooth/a2dp-codec-gst.c
@@ -22,6 +22,7 @@
 #endif
 
 #include <arpa/inet.h>
+#include <stdint.h>
 
 #include <pulsecore/log.h>
 #include <pulsecore/macro.h>
@@ -82,7 +83,7 @@ fail:
 static GstCaps *gst_create_caps_from_sample_spec(const pa_sample_spec *ss) {
     gchar *sample_format;
     GstCaps *caps;
-    int channel_mask;
+    uint64_t channel_mask;
 
     switch (ss->format) {
         case PA_SAMPLE_S16LE:
-- 
GitLab

